agent/internal/agent_test.go

```diff
@@ -0,0 +1,107 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/golang-collections/collections/set"
+)
+
+func TestNoAddProxy(t *testing.T) {
+	inputEnv := container.Config{}
+	testAgent := agent{}
+
+	inputEnv.Env = []string{
+		"FIRST_VAR=1",
+	}
+
+	// InputEnv should not change because we didn't set any environment variables.
+	ans := append([]string{}, inputEnv.Env...)
+
+	testAgent.addProxy(&inputEnv)
+
+	output := set.New()
+	correct := set.New()
+
+	for _, v := range inputEnv.Env {
+		output.Insert(v)
+	}
+
+	for _, v := range ans {
+		correct.Insert(v)
+	}
+
+	if diff := output.Difference(correct); diff.Len() != 0 {
+		t.Errorf("Expected: %v But got: %v", ans, inputEnv.Env)
+	}
```

Comment:
I like using set difference here a lot!  I do think that this whole section could be a helper function now, since it I think it is identical all three times.